src/iotHubResourceExplorer.ts

```diff
@@ -146,15 +151,38 @@ export class IoTHubResourceExplorer extends BaseExplorer {
         if (iotHubItem) {
             outputChannel.show();
             outputChannel.appendLine(`IoT Hub selected: ${iotHubItem.label}`);
-            const iotHubConnectionString = await this.getIoTHubConnectionString(subscriptionItem, iotHubItem.iotHubDescription);
+            const iotHubConnectionString = await this.getIoTHubConnectionString(subscriptionItem.session.credentials,
+                subscriptionItem.subscription.subscriptionId, subscriptionItem.session.environment , iotHubItem.iotHubDescription);
             await this.updateIoTHubConnectionString(iotHubConnectionString);
             (iotHubItem.iotHubDescription as any).iotHubConnectionString = iotHubConnectionString;
             TelemetryClient.sendEvent("AZ.Select.IoTHub.Done", undefined, iotHubConnectionString);
-            await Utility.storeIoTHubInfo(subscriptionItem, iotHubItem.iotHubDescription);
+            await Utility.storeIoTHubInfo(subscriptionItem.subscription.subscriptionId, iotHubItem.iotHubDescription);
             return iotHubItem.iotHubDescription;
         }
     }
 
+    public async setIoTHub(context: IActionContext, node?: IoTHubResourceTreeItem): Promise<void> {
+        if (!node) {
+            node = await this.iotHubTreeDataProvider.showTreeItemPicker<IoTHubResourceTreeItem>("IotHub", context);
+        }
+        this._outputChannel.show();

```

Comment:
the logic here is duplicate with the selectIoTHub, how about abstract a method to share between the commands?

Reply:
updated